ext/ddtrace_profiling_native_extension/extconf.rb

```diff
@@ -125,6 +125,9 @@ def add_compiler_flag(flag)
 # On older Rubies, we need to use a backported version of this function. See private_vm_api_access.h for details.
 $defs << '-DUSE_BACKPORTED_RB_PROFILE_FRAME_METHOD_NAME' if RUBY_VERSION < '3'
 
+# On older Rubies, there are no Ractors
+$defs << '-DNO_RACTORS' if RUBY_VERSION < '3'
```

Comment:
There's no built in `define` around this? Interesting...

Reply:
I did a quick search and there isn't any -- the closest I found has `HAVE_RUBY_RACTOR_H` which can maybe thought of as a way of doing that but... I decided to keep this one for now ;) (Unless you think it's a really bad idea)

I think the reason why there is no define is because you can't disable Ractors (that I know of), and thus you can always probe them by looking at the Ruby version, so that's probably why they don't have a more specific `#define`.

On our side, as you can kinda tell by this file, I've been somewhat avoiding doing something like `$defs << "-DRUBY_#{RUBY_VERSION}"` and then doing `#ifdef RUBY_3.1` in the code, in favor of being a bit more semantic -- e.g. rather than saying "this is the codepath for Ruby 3.1", say "this is the codepath for when this structure has a member called `foo`".